### PR TITLE
feat(manage-variable): create variable Add/Edit/Delete & options validation.. etc

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
SSIA
- Variable Add / Edit / Delete.
- Variable options validation.
- Variable schema properties is replaced to key.
- Interface `DashboardVariableSchemaProperty`'s name is changed to required.

https://user-images.githubusercontent.com/83635051/210315532-1ff9ff38-9332-4ce9-bc8b-deeaf7df2ff1.mov

### CODE REVIEW

**Comment by song2kim** [comment]
image

There are actually two choices, but it seems like there are three because of the '+1' sign, so I think users can get confused.

File: src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue [hunk 1]
```diff
@@ -98,10 +101,10 @@ const state = reactive({
         ADD: i18n.t('DASHBOARDS.CUSTOMIZE.VARIABLES.ADD'),
         EDIT: i18n.t('DASHBOARDS.CUSTOMIZE.VARIABLES.SUB_TITLE_EDIT'),
     })),
-    selected: '',
+    selectedVariable: '',
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue:104]: nitpick: please specify type!

File: src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue [hunk 2]
```diff
@@ -114,21 +117,41 @@ const handleChangeAddContent = () => {
...
     emit('change', properties);
 };
 const handleChangeEditContent = (propertyName: string) => {
-    state.selected = propertyName;
+    state.selectedVariable = propertyName;
     state.contentType = 'EDIT';
 };
+const handleSaveVariable = (variable: DashboardVariableSchemaProperty) => {
+    const properties = cloneDeep(props.variables) as DashboardVariablesSchema['properties'];
+    if (contentType.value === 'ADD') {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue:144]: nitpick: This is more friendly :)

```suggestion
    if (state.contentType=== 'ADD') {
```

File: src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue [hunk 3]
```diff
@@ -98,10 +101,10 @@ const state = reactive({
         ADD: i18n.t('DASHBOARDS.CUSTOMIZE.VARIABLES.ADD'),
         EDIT: i18n.t('DASHBOARDS.CUSTOMIZE.VARIABLES.SUB_TITLE_EDIT'),
     })),
-    selected: '',
+    selectedVariable: '',
 });
 
-const { contentType, titleSet, selected } = toRefs(state);
+const { contentType, titleSet, selectedVariable } = toRefs(state);
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/DashboardManageVariableOverlay.vue:107]: suggestion: How about putting this part at the bottom of the script?

**Review by yuda110 — APPROVED** [review]
LGTM! I only checked conventions.

**Pull request merged**